File: src/buypolarcapital/assets/equities/R/test10.R

source("../../src/data/fetch_data.R")
source("../../src/plotting/plotting.R")

# Define 10 tickers
tickers <- c("AAPL", "MSFT", "GOOG", "AMZN", "TSLA",
             "NVDA", "META", "JPM", "UNH", "NFLX")

# Set time period
start_date <- "2023-01-01"
end_date <- "2023-10-01"

# Prepare output file
output_file <- "plots/buypolar_10_ticker_report.pdf"
if (!dir.exists("plots")) dir.create("plots")
pdf(output_file, width = 10, height = 6)

# Loop through each ticker and generate a plot per page
for (ticker in tickers) {
  cat("Plotting", ticker, "\n")
  df <- get_prices(ticker, start_date = start_date, end_date = end_date)
  
  plot_prices(
    df,
    title = paste(ticker, "Stock Price", format(as.Date(start_date), "%b %Y"), "???", format(as.Date(end_date), "%b %Y")),
    save_pdf = FALSE  # we don't want to export each page again
  )
}

dev.off()
cat("Multi-page PDF saved to:", output_file, "\n")

File: backup_20250731_221529/src/buypolarcapital/assets/equities/python/test10.py
import sys
from pathlib import Path
import os
from matplotlib.backends.backend_pdf import PdfPages
import matplotlib.pyplot as plt

# Add project root to path
sys.path.append(str(Path(__file__).resolve().parents[2]))

from src.data.fetch_data import get_data_yf
from src.plotting.plotting import plot_prices

# Define tickers and date range
tickers = ["AAPL", "MSFT", "GOOG", "AMZN", "TSLA",
           "NVDA", "META", "JPM", "UNH", "NFLX"]

start = "2023-01-01"
end = "2023-10-01"

# Output path
plots_dir = os.path.join(os.getcwd(), "plots")
os.makedirs(plots_dir, exist_ok=True)
pdf_path = os.path.join(plots_dir, "buypolar_10_ticker_report.pdf")

with PdfPages(pdf_path) as pdf:
    for ticker in tickers:
        print(f"📈 Plotting {ticker}...")
        df = get_data_yf(ticker, start=start, end=end)
        
        plot_prices(
            df,
            title=f"{ticker} Stock Price Jan–Oct 2023",
            save_pdf=False,
            export_png=False,
            show=False,  # Don't render live
        )
        
        # Get the actual figure that plot_prices just created
        fig = plt.gcf()
        pdf.savefig(fig)
        plt.close(fig)


print(f"✅ Multi-page PDF saved to: {pdf_path}")
